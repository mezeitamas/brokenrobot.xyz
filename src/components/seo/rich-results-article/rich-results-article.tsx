import React from 'react';
import type { FunctionComponent } from 'react';

type RichResultsArticleProps = {
    siteName: string;
    siteUrl: string;
    title: string;
    url: string;
    published: string;
    author: string;
};

const RichResultsArticle: FunctionComponent<RichResultsArticleProps> = ({
    siteName,
    siteUrl,
    title,
    url,
    published,
    author
}) => {
    return (
        <script type="application/ld+json">
            {`
                {
                    "@context": "https://schema.org",
                    "@type": "BlogPosting",
                    "@id": "${url}#BlogPosting",
                    "headline": "${title}",
                    "name": "${title}",
                    "datePublished": "${published}",
                    "url": "${url}",
                    "author": [{
                        "@type": "Person",
                        "@id": "${siteUrl}/about-me/#Person",
                        "name": "${author}",
                        "url": "${siteUrl}/about-me/"
                    }],
                    "isPartOf": {
                        "@type" : "Blog",
                        "@id": "${siteUrl}",
                        "name": "${siteName}"
                    },
                    "keywords": []
                }
            `}
        </script>
    );
};

export { RichResultsArticle };

export type { RichResultsArticleProps };
