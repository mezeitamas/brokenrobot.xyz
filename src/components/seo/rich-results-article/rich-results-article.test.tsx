import React from 'react';

import { create } from 'react-test-renderer';

import { RichResultsArticle } from './rich-results-article';
import type { RichResultsArticleProps } from './rich-results-article';

describe('<RichResultsArticle />', () => {
    test('should render', () => {
        // Given
        const props: RichResultsArticleProps = {
            siteName: 'Broken Robot',
            siteUrl: 'https://www.brokenrobot.xyz',
            title: 'Hosting a static website on Amazon S3',
            url: 'https://www.brokenrobot.xyz/blog/hosting-a-static-website-on-amazon-s3/',
            published: '2023-05-01T10:10:05.000Z',
            author: 'Tamas Mezei'
        };

        // When
        const component = create(<RichResultsArticle {...props} />).toJSON();

        // Then
        expect(component).toMatchSnapshot();
    });
});
