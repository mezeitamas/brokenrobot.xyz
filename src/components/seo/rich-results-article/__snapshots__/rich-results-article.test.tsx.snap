// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RichResultsArticle /> should render 1`] = `
<script
  type="application/ld+json"
>
  
                {
                    "@context": "https://schema.org",
                    "@type": "BlogPosting",
                    "@id": "https://www.brokenrobot.xyz/blog/hosting-a-static-website-on-amazon-s3/#BlogPosting",
                    "headline": "Hosting a static website on Amazon S3",
                    "name": "Hosting a static website on Amazon S3",
                    "datePublished": "2023-05-01T10:10:05.000Z",
                    "url": "https://www.brokenrobot.xyz/blog/hosting-a-static-website-on-amazon-s3/",
                    "author": [{
                        "@type": "Person",
                        "@id": "https://www.brokenrobot.xyz/about-me/#Person",
                        "name": "Tamas Mezei",
                        "url": "https://www.brokenrobot.xyz/about-me/"
                    }],
                    "isPartOf": {
                        "@type" : "Blog",
                        "@id": "https://www.brokenrobot.xyz",
                        "name": "Broken Robot"
                    },
                    "keywords": []
                }
            
</script>
`;
